Update develop-logseq-on-windows.md to warn about Choco

djm2k opened the pull request:
The scoop `clojure` package being deprecated in favor of `clj-deps` which is already included.

Following up on https://github.com/logseq/logseq/pull/8857:
> - re-add scoop and choco instructions (these are untested)

The choco instructions do not work **for cmd.exe** ; more specifically, the `choco install clojure` command does not provide the `clojure` CLI utility **for cmd.exe**:
  ```
  C:\dev\logseq>yarn watch
  yarn run v1.22.19
  $ run-p gulp:watch cljs:watch
  $ gulp watch
  $ clojure -M:cljs watch app electron
  'clojure' is not recognized as an internal or external command
  ```

After running the the scoop install commands, it works OK:
  ```
  C:\dev\logseq>yarn watch
  yarn run v1.22.19
  $ run-p gulp:watch cljs:watch
  $ gulp watch
  $ clojure -M:cljs watch app electron
  Downloading: org/clojure/clojurescript/1.11.54/clojurescript-1.11.54.pom from central
  Downloading: cider/cider-nrepl/0.29.0/cider-nrepl-0.29.0.pom from clojars
  Downloading: org/clojars/knubie/cljs-run-test/1.0.1/cljs-run-test-1.0.1.pom from clojars
  [11:19:49] Using gulpfile C:\dev\logseq\gulpfile.js
  [11:19:49] Starting 'watch'...
  ```

Looks like scoop adds a `clojure.exe` shim, which the choco package does not:
![image](https://github.com/djm2k/logseq/assets/36978885/40d5d5e4-2916-4e6d-a4c5-ba9036cc8db0)

andelf commented:
Choco installs Clojure as a PowerShell module, where `clojure/clj` acts as an alias for Invoke-Clojure. This installation method is consistent with the official Clojure on Windows release.

Regarding the preferred way for Windows developers, it's unclear to me. 
Personally, I have managed this by creating scripts such as clojure.bat or clojure.ps1.
Additionally, I write a toy project https://github.com/andelf/clojure-cli that adds an exe wrapper for the underlying java command.

I prefer to add a detailed explanation for why the official "clj on Windows" and the "Choco" version does not work.

CC @cnrpman

djm2k commented:
> Choco installs Clojure as a PowerShell module, where `clojure/clj` acts as an alias for Invoke-Clojure.

Thank you for this! I only tried running cmd.exe as administrator so couldn't figure out what that package was supposed to do; although I am fairly impatient when it comes to package managers...

I have moved Scoop to the top and added explainers about this issue under Scoop and Choco, as well as your binary.

andelf reviewed (approved):
LGTM. Many thanks.
